### PR TITLE
feat: introduce semantic release

### PR DESCRIPTION
**Pull request by blimmer**
This PR introduces semantic-release so it's easy to automatically release new versions of this package.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: 2.1
 orbs:
   node: circleci/node@5.0.0
 workflows:
-  test:
+  build-and-deploy:
```
Comment (blimmer) [inline .circleci/config.yml:5]: Here's what this workflow will look like:

Screen Shot 2022-02-09 at 18 00 29

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,9 +12,18 @@ workflows:
...
           matrix:
+            alias: Run Tests
             parameters:
               version:
                 - "12.22"
                 - "14.19"
                 - "16.13"
                 - "lts"
+          requires: ["Validate Build", "Lint"]
+      - node/run:
```
Comment (blimmer) [inline .circleci/config.yml:23]: @stephenh , it looks like there isn't a CircleCI context set up for the `stephenh` organization in CircleCI. Could you either:

a) create one for NPM publishing, or
b) manually add `GITHUB_TOKEN` and `NPM_TOKEN` to this single project

The docs on what those tokens need to be are described here: https://semantic-release.gitbook.io/semantic-release/usage/ci-configuration#authentication

File: .github/workflows/conventional-commit-pr-title.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Validate PR Title
```
Comment (blimmer) [inline .github/workflows/conventional-commit-pr-title.yml:1]: Will you also enforce "Squash and Merge" in the repo settings?

Comment (stephenh) [reply .github/workflows/conventional-commit-pr-title.yml:1]: Done

File: .circleci/config.yml (hunk 2 above)
Comment (stephenh) [reply .circleci/config.yml:23]: Done, hopefully

**Pull request merged**